src/Centreon/Domain/PlatformTopology/PlatformTopology.php

```diff
@@ -0,0 +1,234 @@
+     * @throws PlatformTopologyException
+     */
+    public function setServerType(int $serverType): self
+    {
+        // The API should not be used to add a Central to another Central
+        if (self::SERVER_TYPE_CENTRAL === $serverType) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("You cannot link the Central '%s'@'%s' to another Central"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        // Check if the server_type is available
+        $availableServerType = [
+            self::SERVER_TYPE_POLLER,
+            self::SERVER_TYPE_REMOTE,
+            self::SERVER_TYPE_MAP,
+            self::SERVER_TYPE_MBI
+        ];
+        if (!in_array($serverType, $availableServerType)) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("The type of platform '%s'@'%s' is not consistent"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->serverType = $serverType;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServerName(): string
+    {
+        return $this->serverName;
+    }
+
+    /**
+     * @param string $serverName
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerName(string $serverName): self
+    {
+        $serverName = filter_var($serverName, FILTER_SANITIZE_STRING);
+        if (empty($serverName)) {
+            throw new PlatformTopologyException(
+                _('The name of the platform is not consistent'),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->serverName = $serverName;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServerAddress(): string
+    {
+        return $this->serverAddress;
+    }
+
+    /**
+     * @param string $serverAddress
+     *
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerAddress(string $serverAddress): self
+    {
+        $this->serverAddress = $this->checkIpAddress(self::KIND_SERVER, $serverAddress);
+        return $this;
+    }
+
+    /**
+     * Validate address consistency
+     *
+     * @param string $kind server or parent of the server
+     * @param string|null $address the address to be tested
+     *
+     * @return string
+     * @throws PlatformTopologyException
+     */
+    private function checkIpAddress(string $kind, ?string $address): string
+    {
+        // Server linked to the Central, may not send a parent address in the data
+        if (!isset($address) && self::KIND_PARENT === $kind) {
+            return $_SERVER['SERVER_ADDR'];
```

Comment:
What happens if you are not in the context of a web application?
$_SERVER['**SERVER_ADDR**'] is for context of web application.

Reply:
This endpoint can only be called on this context on Central or Remote platform.
What do you propose to get the local platform IP when the API is processed ?